pkg/catalogv2/system/system.go

```diff
@@ -367,24 +372,26 @@ func (m *Manager) isInstalled(namespace, name, version, minVersion string, desir
 		return false, "", nil, err
 	}
 
-	desired, err := semver.NewVersion(version)
-	if err != nil {
-		return false, "", nil, err
-	}
+	return isInstalled(releases, minVersion, version, desiredValue)
+}
 
-	for _, release := range releases {
-		if release.Info.Status != release2.StatusDeployed {
+func isInstalled(releases []*release.Release, minVersion, latestVersion string, desiredValues map[string]any) (bool, string, map[string]interface{}, error) {
+	if minVersion == "" && latestVersion == "" {
```

Comment:
Two things: explicit requirements for the inputs and performance.
Without the check, the behavior will be the same. The function will just throw an error later when attempting to parse the empty string as one denoting some semantic version. We don't need to let it go that far and should just check the basic case first.

Reply:
> explicit requirements for the inputs and performance.

IMO, the `latestVersion` parameter isn't controlled by the user, so having an explicit requirement for this value doesn't make too much sense. Since this value is derived from the chart itself, there's various reasons (some of which are in the calling function's control control, some of which aren't) which could be causing an issue here. 

If we want a requirement for this, I think it would best go in Ensure so that the function which calls ensure get's an error that it supplied invalid input.